File: 09-08-2024_Maximize Array Value After Rearrangement.java


class Solution {

    int Maximize(int arr[]) {
        int n = arr.length;
        int mod = 1000000007;

        // Sorting the array in ascending order
        Arrays.sort(arr);

        long ans = 0;

        // Calculating the sum of product of each element with its index
        // and taking modulo to avoid overflow
        for (int i = 0; i < n; i++) ans = (ans + ((long)arr[i] * i) % mod) % mod;

        return (int)ans;
    }
}

File: 09-08-2024_Maximize Array Value After Rearrangement.py

class Solution:

    #Function to maximize the sum
    #by rearranging the array elements.
    def Maximize(self, arr):

        n = len(arr)
        arr.sort()  #sorting the array

        mod = 1000000007  #taking modulo for large values

        s = 0  #variable to store the sum

        for i in range(n):
            s += arr[
                i] * i  #calculating the sum of array elements multiplied by their index
            s = s % mod  #taking modulo to avoid large sum value

        return s % mod  #returning the final sum after taking modulo.
